### PR TITLE
Add device view with direct node selection

### PR DESCRIPTION
**Pull request by triwav**

### CODE REVIEW

File: src/managers/RtaManager.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            this.lastStoreNodesResponse = await rta.odc.storeNodeReferences(args, options);
+
+            const viewIds = [];
+            if (requestorId === 'rokuDeviceView') {
+                viewIds.push('sceneGraphInspectorView');
+            } else if (requestorId === 'sceneGraphInspectorView') {
+                viewIds.push('rokuDeviceView');
+            }
+            this.webviewViewProviderManager.sendMessageToWebviews(viewIds, {
+                event: 'storedNodeReferencesUpdated'
```
Comment (triwav) [inline src/managers/RtaManager.ts:81]: Try to make these an enum as well

File: src/viewProviders/BaseWebviewViewProvider.ts [hunk 2]
```diff
@@ -81,11 +105,12 @@ export abstract class BaseWebviewViewProvider implements vscode.WebviewViewProvi
...
     protected async getHtmlForWebview() {
         try {
-            if (util.isExtensionHostRunning() && !this.outDirWatcher) {
+            const watcherPath = require.resolve('@parcel/watcher');
+            if (watcherPath && !this.outDirWatcher) {
                 // When in dev mode, spin up a watcher to auto-reload the webview whenever the files have changed.
                 // eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
                 this.outDirWatcher = await require('@parcel/watcher').subscribe(this.webviewBasePath, (err, events: Event[]) => {
-                    //only refresh when the index.html page is changed. Since vite rewrites the file on every cycle, this is enough to know to reload the page
+                    //only refresh when the index.html page is changed. Since svelte rewrites the file on every build, this is enough to know to reload the page
```
Comment (triwav) [inline src/viewProviders/BaseWebviewViewProvider.ts:113]: put back to vite

File: webviews/src/views/RokuDeviceView/RokuDeviceView.svelte [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+    }
+
+    async function findNodesAtLocation(args) {
+        try {
+            const result = await odc.findNodesAtLocation(args);
+            focusedNodeTree = result.matches[0];
+        } catch(e) {
+            console.log(e);
+
+            debugger;
```
Comment (triwav) [inline webviews/src/views/RokuDeviceView/RokuDeviceView.svelte:164]: remove debugger

File: webviews/src/views/RokuDeviceView/RokuDeviceView.svelte [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+    }
+
+    let imageWidth = 0;
+    let imageHeight = 0;
+
+
+    let imageSizeAdjust = 1;
+    $: {
+        if (imageWidth) {
+            imageSizeAdjust = imageWidth / 1920
```
Comment (triwav) [inline webviews/src/views/RokuDeviceView/RokuDeviceView.svelte:57]: ;
```suggestion
            imageSizeAdjust = imageWidth / 1920;
```

File: webviews/src/views/RokuDeviceView/RokuDeviceView.svelte [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+    });
+
+    let screenshotOutOfDateTimeOut;
+    let currentlyCapturingScreenshot = false;
+    async function requestScreenshot() {
+        if (!enableScreenshotCapture || currentlyCapturingScreenshot) {
+            return;
+        }
+        currentlyCapturingScreenshot = true;
+        const {success, arrayBuffer} = await intermediary.sendMessage('getScreenshot') as any;
```
Comment (triwav) [inline webviews/src/views/RokuDeviceView/RokuDeviceView.svelte:227]: Switch to try finally to set back

File: webviews/src/views/SceneGraphInspectorView/NodeBranchPage.svelte [hunk 6]
```diff
@@ -1,4 +1,5 @@
 <script lang="ts">
+    // TODO this probably should be named somethings besides a page
```
Comment (triwav) [inline webviews/src/views/SceneGraphInspectorView/NodeBranchPage.svelte:2]: rename to nodebranch

**Review by TwitchBronBron — APPROVED** [review]
We reviewed this together on a video call, and it looks great!

**Pull request merged**